scheduler/src/cook/mesos/monitor.clj

```diff
@@ -53,32 +47,32 @@
                        stats (-> job-ent
                                  util/job-ent->resources
                                  (select-keys [:cpus :mem])
-                                 (assoc :count 1))]
+                                 (assoc :jobs 1))]
                    {user stats})))
          (reduce (partial merge-with (partial merge-with +)) {}))))
 
-(defn add-aggregated-stats
+(defn- add-aggregated-stats
   "Given a map from users to their stats, associcate a special user
    \"all\" for the sum of all users stats."
   [db stats]
   (if (seq stats)
     (->> (vals stats)
          (apply merge-with +)
          (assoc stats "all"))
-    {"all" (zipmap (conj (util/get-all-resource-types db) :count)
-                   (repeat 0.0))}))
+    {"all" {:cpus 0, :mem 0, :jobs 0}}))
 
-(defn get-starved-job-stats
+(defn- get-starved-job-stats
```

Comment:
Should we run this and see if there are any effects on performance due to the additional datomic queries via `get-share`?

Reply:
Thanks for pointing this out. I changed this to do a single datomic query using `get-shares` for all of the waiting users, instead of doing a query for each individual waiting user